### DISCUSSION TITLE
How to trigger change event when the Input range value changed programmatically with Alpine.js

### DISCUSSION THREAD

**Discussion by mir161803** [alpinejs/alpine#3320] [1. Help] [answered]
So i have the input range that changes x-data value when change happen (I've bind the it with x-model.data)
```
x-data="{data: '3', yearly: false, pageviews: 100, price: '$16.00', yearlyPrice: '$12.00'}
```
the objective is to change price and yearly price and yearlyPrice so I use @change on input range like below
```
@change="price = data === '1' ? '$8.00': data === '2' ? '$12.00' : data === '3' ? '$16.00' : data === '4' ? '$24.00' : '$36.00'; yearlyPrice = data === '1' ? '$6.00' : data === '2' ? '$9.00' : data === '3' ? '$12.00': data === '4' ? '$18.00': '$27.00'"
```

when I test it manually using mouse cursor it works perfectly, but when I use inspect element console and try to changed it programmatically like below, the slider cursor does change but the @change code is not executed
```
slider.value='4'
```

So how do we make on change could listen when we change it programmatically? (like change it via inspect element console)

**Comment by SimoTod** [comment]
I feel like it's more complicated than it should be, i would not use events. If they are always functions of data, use computed properties.
Changing an input via dev console is not a real use case. In alpine you shouldn't manipulate the dom but the component state and let the dom react. If your slider is modelling a property (data?), just update that property.

**Comment by ekwoka** [comment, marked as answer]
As mentioned above, you shouldn't be interacting with the range in that way.

Also, it would likely be smarter to make those objects an object so you can just do `prices[value]`

But most specifically, you model the `value` to state, and the price is more like
```js
get price() {
  return prices[this.value]
}
```

**Reply by mir161803** [reply]
wow amazing, thank you @SimoTod @ekwoka, hence we should be define all possible state and just just changing it in dom. is it correct?
